latch_cli/docker_utils/__init__.py

```diff
@@ -0,0 +1,212 @@
+from dataclasses import dataclass
+from enum import Enum, auto
+from pathlib import Path
+from textwrap import dedent
+from typing import List
+
+import yaml
+
+from latch_cli.constants import latch_constants
+from latch_cli.types import LatchWorkflowConfig
+
+
+class DockerCmdBlockOrder(Enum):
+    FIRST = auto()
+    PRECOPY = auto()
+    POSTCOPY = auto()
+    LAST = auto()
+
+
+@dataclass
+class DockerCmdBlock:
+    comment: str
+    commands: List[str]
+    order: DockerCmdBlockOrder
+
+
+def get_prologue(config: LatchWorkflowConfig) -> List[str]:
+    return [
+        "# latch base image + dependencies for latch SDK -- removing these will break the workflow",
+        f"from {config.base_image}",
+        f"run python3 -m pip install latch=={config.latch_version}",
+    ]
+
+
+def get_epilogue() -> List[str]:
+    return [
+        "# latch internal tagging system + expected root directory -- changing these lines will break the workflow",
+        "arg tag",
+        "env FLYTE_INTERNAL_IMAGE $tag",
+        "WORKDIR /root",
+    ]
+
+
+def infer_commands(pkg_root: Path) -> List[DockerCmdBlock]:
+    commands: List[DockerCmdBlock] = []
+
+    # apt requirements in workflow directory
+    if (pkg_root / "requirements.apt").exists():
+        print("Install apt requirements from `requirements.apt`")
+        commands.append(
+            DockerCmdBlock(
+                comment="install apt requirements",
+                commands=[
+                    "copy requirements.apt /root/requirements.apt",
+                    "run apt-get update -y && xargs apt-get install -y </root/requirements.apt",
+                ],
+                order=DockerCmdBlockOrder.PRECOPY,
+            )
+        )
+
+    # R requirements in workflow directory
+    if (pkg_root / "environment.R").exists():
+        print("Install R packages from `environment.R`y")
+        commands.append(
+            DockerCmdBlock(
+                comment="install R requirements",
+                commands=[
+                    dedent(
+                        """\
```

Comment:
```suggestion
                        """
```
?
if you need newlines, maybe include them when joining the blocks?

Reply:
consider using an r-string to get rid of all the `\\` escapes

Reply:
oh yes this is good